runtime-modules/council/src/lib.rs

```diff
@@ -717,22 +726,53 @@ decl_module! {
         #[weight = 10_000_000] // TODO: adjust weight
         pub fn funding_request(
             origin,
-            amount: Balance::<T>,
-            account: T::AccountId,
+            funding_requests: Vec<FundingRequestParameters<Balance<T>, T::AccountId>>
         ) {
             // Checks
             ensure_root(origin)?;
+
+            let funding_total: Balance<T> =
+                funding_requests.iter().fold(
+                    Zero::zero(),
+                    |accumulated, funding_request| accumulated + funding_request.amount,
+                );
+
             let current_budget = Self::budget();
-            ensure!(amount<=current_budget, Error::<T>::InsufficientFundsForFundingRequest);
 
+            ensure!(
+                funding_total <= current_budget,
+                Error::<T>::InsufficientFundsForFundingRequest
+            );
+
+            ensure!(!funding_requests.is_empty(), Error::<T>::EmptyFundingRequests);
 
+            let mut recieving_accounts = Vec::<&T::AccountId>::new();
+
+            for funding_request in &funding_requests {
+                ensure!(
+                    funding_request.amount != Zero::zero(),
+                    Error::<T>::ZeroBalanceFundRequest
+                );
+
+                ensure!(
+                    !recieving_accounts.contains(&&funding_request.account),
+                    Error::<T>::RepeatedFundRequestAccount
+                );
+
+                recieving_accounts.push(&funding_request.account);
+            }
             //
             // == MUTATION SAFE ==
             //
 
-            Mutations::<T>::set_budget(&(current_budget - amount));
-            let  _ = balances::Module::<T>::deposit_creating(&account, amount);
-            Self::deposit_event(RawEvent::RequestFunded(account, amount));
+            for funding_request in funding_requests {
+                let amount = funding_request.amount;
+                let account = funding_request.account;
+                let current_budget = Self::budget();
```

Comment:
You have calculated the total balance to subtract from the budget earlier. Can we make it a single call prior to the cycle?